pytools/graph.py

```diff
@@ -232,7 +239,38 @@ def compute_topological_order(graph):
                 # so either this is a leaf or all children have been visited
                 visiting.remove(node)
                 visited.add(node)
-                reverse_order.append(node)
+
+                if key is None:
+                    reverse_order.append(node)
+                else:
+                    # {{{ move every 'u' from heap to reverse_order s.t. 'node'->'u'
+
+                    nodes_to_move = set(graph.get(node, ())) & nodes_in_heap
+
+                    while nodes_to_move:
+                        node_to_move = nodes_to_move.pop()
+
+                        while True:
+                            _, popped_node = heappop(heap)
+                            reverse_order.append(popped_node)
+                            nodes_to_move.discard(popped_node)
+                            nodes_in_heap.remove(popped_node)
+                            if popped_node == node_to_move:
+                                break
```

Comment:
What is the situation here?

Reply:
Since `node` is ready to be scheduled i.e. its time to push `node` into `heap`. `heap` cannot contain any immediate successors of `node` as its only job is to help maintaining  priorities but not in a way that invalidates the topological sort order. So all the immediate successors of `node` that are present the `heap` must be completely scheduled (i.e. moved to reverse_order).

The job of the while-loop+heap is to completely schedule all the nodes that are at lesser priority than the successors of `node` in that order.

Reply:
(Don't explain these things in PR comments. Add actual code comments.)

Reply:
Resolving as that part of code no longer exists.